clouddriver-kubernetes/src/integration/java/com/netflix/spinnaker/clouddriver/kubernetes/it/utils/KubeTestUtils.java

```diff
@@ -355,4 +348,54 @@ public static void deployAndWaitStable(
             .asList();
     KubeTestUtils.deployAndWaitStable(baseUrl, body, namespace, kind + " " + name);
   }
+
+  public static void deployIfMissing(
+      String baseUrl,
+      String account,
+      String namespace,
+      String kind,
+      String name,
+      String app,
+      KubernetesCluster kubeCluster)
+      throws InterruptedException, IOException {
+    deployIfMissing(baseUrl, account, namespace, kind, name, app, null, kubeCluster);
+  }
+
+  public static void deployIfMissing(
+      String baseUrl,
+      String account,
+      String namespace,
+      String kind,
+      String name,
+      String app,
+      String image,
+      KubernetesCluster kubeCluster)
+      throws InterruptedException, IOException {
+
+    String path = "";
+    if (image != null) {
+      path = ".spec.template.spec.containers[0].image";
+    }
+
+    String output =
+        kubeCluster.execKubectl(
+            "-n "
+                + namespace
+                + " get "
+                + kind
+                + " -l app.kubernetes.io/name="
+                + app
+                + " -o=jsonpath='{.items[?(@.metadata.name==\""
+                + name
+                + "\")]"
+                + path
+                + "}'");
+    if (!output.isEmpty()) {
+      if (image == null || output.contains(image)) {
+        return;
+      }
+    }
+
+    deployAndWaitStable(baseUrl, account, namespace, kind, name, app, image);
```

Comment:
Do you think we should also support ```
deployAndWaitStable(
      String baseUrl, List<Map<String, Object>> reqBody, String targetNs, String... objectNames)
``` signature method, or to have a separate one that does? Since it is much more flexible.

Reply:
We can add it when we need it, today it wouldn't be used